### ISSUE TITLE
Page Title and Template Changes After Saving page on Elementor Builder

### ISSUE THREAD

**Issue by kikoking1** [pojome/elementor#4186]
## Description

On my home page (front page) in Wordpress, the title and chosen template change every time I save when I am in the Elementor builder for that page. It takes on the same chosen template and title as one of my other blog posts. This is almost making the page builder impractical for us to use on this project. It is extremely annoying and hindering on my client's site. Please help me!

A note: the system info complains that my memory limit is set to 40M, but I have confirmed that it's actually 256M.

## Steps to reproduce

1. Save home page title as "Home" in title field and save Template Elementor Full Width
2. Click Edit with Elementor on home page
3. Click update button to save Elementor page builder changes.
4. Go back to home page to see the title and template have both changed, reflecting that of one of the blog posts.

## Environment
					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: 5.5.5
	PHP Version: 7.0.29
	PHP Max Input Vars: 1500
	PHP Max Post Size: 200M
	GD Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 4.9.5
	Site URL: *****
	Home URL: *****
	WP Multisite: No
	Max Upload Size: 200 MB
	Memory limit: 40M
	Permalink Structure: /%category%/%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Astra Child
	Version: 1.0.0
	Author: Brainstorm Force
	Child Theme: Yes
	Parent Theme Name: Astra
	Parent Theme Version: 1.2.7.2
	Parent Theme Author: Brainstorm Force

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36

== Active Plugins ==
	Advanced Custom Fields
		Version: 4.4.12
		Author: Elliot Condon

	Advanced Custom Fields PRO
		Version: 5.6.10
		Author: Elliot Condon

	Custom Post Type UI
		Version: 1.5.8
		Author: WebDevStudios

	Elementor
		Version: 2.0.7
		Author: Elementor.com

	Elementor Pro
		Version: 2.0.2
		Author: Elementor.com

	Redirection
		Version: 3.2
		Author: John Godley

	Remove Category URL
		Version: 1.1.2
		Author: Valerio Souza

	The Events Calendar
		Version: 4.6.14.1
		Author: Modern Tribe, Inc.

	The Events Calendar Shortcode
		Version: 1.8
		Author: Event Calendar Newsletter

	Yoast SEO
		Version: 7.3
		Author: Team Yoast

== Debug ==
	Errors: 
		ReferenceError: hbspt is not defined
			at /wp-admin/post.php?post=14&action=elementor - 2:8
			2018-04-20 21:39 +00:00

		ReferenceError: hbspt is not defined
			at /wp-admin/post.php?post=14&action=elementor - 4:8
			2018-04-20 21:39 +00:00

		ReferenceError: hbspt is not defined
			at /wp-admin/post.php?post=14&action=elementor - 2:8
			2018-04-20 21:28 +00:00

		ReferenceError: hbspt is not defined
			at /wp-admin/post.php?post=14&action=elementor - 4:8
			2018-04-20 21:28 +00:00

		ReferenceError: hbspt is not defined
			at /wp-admin/post.php?post=14&action=elementor - 2:8
			2018-04-20 21:24 +00:00

		ReferenceError: hbspt is not defined
			at /wp-admin/post.php?post=14&action=elementor - 4:8
			2018-04-20 21:24 +00:00

		ReferenceError: hbspt is not defined
			at /wp-admin/post.php?post=14&action=elementor - 2:8
			2018-04-20 21:15 +00:00

		ReferenceError: hbspt is not defined
			at /wp-admin/post.php?post=14&action=elementor - 4:8
			2018-04-20 21:15 +00:00

		ReferenceError: hbspt is not defined
			at /wp-admin/post.php?post=14&action=elementor - 2:8
			2018-04-20 21:14 +00:00

		ReferenceError: hbspt is not defined
			at /wp-admin/post.php?post=14&action=elementor - 4:8
			2018-04-20 21:14 +00:00

**Comment by notrealdev** [comment]
Same problem here, i deleted and reinstall Elementor
Then I did not issue this again, I guess that in your theme there are at least 1 template

**Comment by kikoking1** [comment]
I think I fixed the issue! I have created a custom plugin for my clients site that added a shortcode which I added to the page builder page, which outputs a bunch of posts from a custom post type. At the end I call wp_reset_query() in the shortcode function to reset the custom post loop. Turns out when I remove just that one line, everything is fixed! No idea why this happens. If someone can help shed light maybe I can keep the code more clean to avoid this issue with Elementor/Wordpress.

**Comment by bainternet** [comment]
@kikoking1 

Probably, your Query was messing up the global $post so the editor couldn't save correctly.

glad you figured it out.

**Comment by masterbip** [comment]
Hi there. Same thing here. Triying to create a shortcode to get products by category. I am not an expert, by the way.

This is the code

`function register_shortcodes(){
	add_shortcode('productos','shortcode_productos');
	add_shortcode('mbsocial','shortcode_redes_sociales');
}
add_action( 'init', 'register_shortcodes');

/* Blog */
function shortcode_productos($atts){
	$atts = shortcode_atts(
		array(
			'categoria' => ''
		), $atts);
	
	$categoria = $atts['categoria'];
	
	$contador = 1;

	query_posts(array(
		'post_type' => 'product',
		'posts_per_page' => -1,
		'order' => 'DESC',
		'orderby' => 'date',
		'tax_query' => array(
			array(
				'taxonomy' => 'product_cat',
				'field' => 'slug',
				'terms' => $categoria,
				'operator' => 'IN'
			)
		)
	));
	$return_string = '<ul class="productos-shortcode">';

	if (have_posts()) :

	while (have_posts()) : the_post();
	
	global $post;
	global $product;

	$thumb = get_the_post_thumbnail_url($post->ID,'medium');

	/* REPEATER */
	$return_string .= '<li id="producto_'.get_the_ID().'" class="producto-shortcode-item producto-0'.$contador.' producto-id-'.get_the_ID().'">';
	$return_string .= '<div class="producto-shortcode-wrap transicion">';
	//$return_string .= '<a href="'.get_the_permalink().'" class="producto-contenido">';
	$return_string .= '<img src="'.$thumb.'" />';
	$return_string .= '<h2>'.get_the_title().'</h2>';
	$return_string .= wpautop(get_the_content());
	$return_string .= '<span class="precio">'.$product->get_price_html().'</span>';
	//$return_string .= '</a>';
	$return_string .= '</div>';
	$return_string .= '</li>';
	/* REPEATER */

	$contador = $contador + 1;

	endwhile;
	endif;

	$return_string .= '</ul>';

	wp_reset_query();

	return $return_string;
}`

If i delete this shortcode, all is fine.
If i use this, I got changed page Title and elementor don't see any templates I got.
If I delete "wp_reset_query();" from the shortcode I see a message ina modal: 
"Lo sentimos, no se encontró el área de contenido en su página.
Debes llamar a la función 'the_content' en la plantilla actual, para que Elementor pueda funcionar en esta página."
Something like "Sorry, content area was not found, You should call "the_content" function...."

And my template page is so simple:
`<?php 
/*
Template name: Inicio
*/

get_header();

while ( have_posts() ) : the_post();
?>
<section>
	<div class="container container-full">
		<div class="contenido">
			<?php the_content(); ?>
		</div>
	</div>
</section>
<?php endwhile; ?>
<?php get_footer(); ?>`

You see that "the_content" is there. So i assume elementor  can't see any page template at all, not even elementor default ones ...

So .... What is wrong with my shortcode?
Because I think that is the problem....

**Comment by masterbip** [comment]
It is always good to write a question for me ... it makes me think twice.
So ... i rewrite my hole shortcode. And it is all working fine.
If anyone need it this is the costom woocommerce shortcode to get products by category, and works perfect with elementor (Did I said that elementor is like ... the "santo grial" for us?
----------------------------
``
/* Productos */
function shortcode_productos($atts){
	$atts = shortcode_atts(
		array(
			'categoria' => ''
		), $atts);
	
	$categoria = $atts['categoria'];
	
	$contador = 1;
	
	$args = array(
		'post_type' => 'product',
		'posts_per_page' => -1,
		'order' => 'DESC',
		'orderby' => 'date',
		'tax_query' => array(
			array(
				'taxonomy' => 'product_cat',
				'field' => 'slug',
				'terms' => $categoria,
				'operator' => 'IN'
				)
			)
		);
		$return_string = '<ul class="productos-shortcode">';
		$loop = new WP_Query( $args );
		if ( $loop->have_posts() ) {
			while ( $loop->have_posts() ) : $loop->the_post();
				global $product;
				$thumb = get_the_post_thumbnail_url($post->ID,'medium');
				$return_string .= '<li id="producto_'.get_the_ID().'" class="producto-shortcode-item producto-0'.$contador.' producto-id-'.get_the_ID().'">';
				//wc_get_template_part( 'content', 'product' );
				$return_string .= '<div class="producto-shortcode-wrap transicion">';
				//$return_string .= '<a href="'.get_the_permalink().'" class="producto-contenido">';
				$return_string .= '<img src="'.$thumb.'" />';
				$return_string .= '<h2>'.get_the_title().'</h2>';
				$return_string .= wpautop(get_the_content());
				$return_string .= '<span class="precio">'.$product->get_price_html().'</span>';
				//$return_string .= '</a>';
				$return_string .= '</div>';
				$return_string .= '</li>';
			endwhile;
		} else {
			$return_string .= __( 'No products found' );
		}
	
	wp_reset_postdata();
	
	$return_string .= '</ul>';
	
	return $return_string;
}
``

----------------------------

**Comment by jmosheim** [comment]
I've spent many hours trying to make simple changes in a template to the spacing in the pages without success through Elementor/WP.  Just by looking at the comments above, to have to write that level of code as a patch to do a not very complex change is baffling to me.  If I would have to start all over again, I would forget about Elementor, learn php basics and use the WP editor and write the pages from scrtch - it saves ++++ time at the end, it would have saved me lots of hours of frustration.  I found the "culprit" code through Google Dev in a snap, not complicated at all...but WP/Elementor has it buried in a solid one page of continuous characters of unstructure code that is impossible to find unless you have nothing to do one Sunday afternoon... I think php seems to be a good language but WP is way over hyped. 

**For non php programmers like me, **STAY AWAY FROM WP**, stick w HTML & Javascript.** 

Just look at the equivalent length of code of two similar pages one with WP and another one w HTML/JS and its baffling, the WP one is like 5 times longer and doing the same thing.
